internal/signaling/timeout_manager.go

```diff
@@ -38,51 +30,57 @@ func (i *TimeoutManager) Run(ctx context.Context) {
 func (i *TimeoutManager) RunOnce(ctx context.Context) {
 	logger := logging.GetLogger(ctx)
 
-	i.Lock()
-	defer i.Unlock()
-
-	now := util.Now(ctx)
-	for _, tp := range i.peers {
-		p := tp.peer
-		t := tp.time
-		if now.Sub(t) > i.DisconnectThreshold {
-			logger.Debug("peer timed out closing peer", zap.String("id", p.ID))
-			delete(i.peers, p.ID+p.Secret)
-			go p.Close(ctx)
+	for ctx.Err() == nil {
+		hasNext, err := i.Store.ClaimNextTimedOutPeer(ctx, i.DisconnectThreshold, func(peerID, gameID string, lobbies []string) error {
+			logger.Debug("peer timed out closing peer", zap.String("id", peerID))
+
+			for _, lobby := range lobbies {
+				packet := DisconnectPacket{
+					Type: "disconnect",
+					ID:   peerID,
+				}
+				data, _ := json.Marshal(packet)
+				ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+				defer cancel()
```

Comment:
Defer in a loop (sorry for the interruption 🙈 )

Reply:
Woops! I completely missed your comment before merging. My bad. Nice catch! Will quickly fix it in a new PR. 👍🏻

Reply:
Here is the fix: https://github.com/poki/netlib/pull/36

Thanks for the interruption! 👍🏻